jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -670,14 +667,33 @@ YUI.add('user-profile', function() {
     },
 
     /**
+      Handle the destroy model button being clicked.
+
+      @method _handleDestroyModel
+    */
+    _handleDestroyModel: function() {
+      var model = this.state.destroyModel;
+      var uuid = model.uuid;
+      // Hide the confirmation popup.
+      this._displayConfirmation(null);
+      // If the model to be distroyed is active then we can destroy it
+      // immediately.
+      if (uuid === this.props.currentModel) {
```

Comment:
I've added a comment. I'm not sure what we can do there except may surface errors about not having switched models correctly.